### PR TITLE
Kubernetes cluster type 'gke' not respecting backend auth

### PR DESCRIPTION
**Pull request by armandocomellas1**
#### :heavy_check_mark: Checklist

- [x] ChangeSet name is thirty-sloths-sin.md and there l describe what it doing it the code. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

The implementation consist in adding a piece of code starts in 58 adding some config.getString() method to find out if the app-config.yaml kubernetes config includes the type of Auth as *googleServiceAccount*

So the behavior is:

Before implementation we had this:
image

After implementarion:
image

So as you can see after the implementation the UI renders all the pods related to the Kubernetes Engine Clusters

Testing the script with the implementation:
image

Thank you for reviewing the pull request.

### CODE REVIEW

File: plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts
Comment (mclarke47) [inline plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts:61]: Could this try catch be

`config.GetStringOptional('authProvider') ?? "google"`

File: plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts
Comment (jamieklassen) [inline plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts:69]: I might be refactoring too aggressively here:
```suggestion
    const storeAuthProviderString = config.getOptionalString('authProvider') === 'googleServiceAccount' ? 'googleServiceAccount' : 'google';
```

regardless, can we have some unit tests for this (they live [here](https://github.com/backstage/backstage/blob/master/plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.test.ts))? I think we'd want at least scenarios for:

* no `authProvider` specified
* `authProvider = 'googleServiceAccount'`
* `authProvider` is neither `googleServiceAccount` nor `google`

Furthermore, let's update the config schema here to mention this new parameter:

https://github.com/backstage/backstage/blob/2d59a19fd3a566dbb8f1639aab8c79f84db7db2a/plugins/kubernetes-backend/config.d.ts#L76-L87

File: .changeset/many-pugs-change.md
Comment (jamieklassen) [inline .changeset/many-pugs-change.md:5]: The purpose of these changesets is not really to describe the code change to developers; they get used to generate release notes for end users (see some [advice](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#writing-changesets) in `CONTRIBUTING.md`)

Can you change this to be a description for end-users of what the new feature is and how to use it?

**Review by jamieklassen — CHANGES_REQUESTED** [review]
Thanks for opening this, I think I never saw #13648! Comments inline. Important stuff: changeset, unit tests and config schema.

File: plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts
Comment (mclarke47) [reply plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts:61]: I would use the version suggested by @jamieklassen below which aims to resolve the same thing

Comment (armandocomellas1) [reply plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts]: @mclarke47 Done!

Thanks

File: plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts
Comment (armandocomellas1) [reply plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts]: @jamieklassen  Better way to simplify!

Thank you!

File: .changeset/many-pugs-change.md
Comment (armandocomellas1) [reply .changeset/many-pugs-change.md]: Duley noted!

Added better description to being understandable for the end user.

Thank  you!

**Comment by armandocomellas1** [comment]
Hi guys!

@jamieklassen @mclarke47 @erikprice Thisi is just a reminder that we have this PR pending.

Thank you for taking the time to reviewing it.

**Review by jamieklassen — CHANGES_REQUESTED** [review]
Changeset looks better and unit tests look pretty good; can you also update the config schema here to mention this new parameter?

https://github.com/backstage/backstage/blob/2d59a19fd3a566dbb8f1639aab8c79f84db7db2a/plugins/kubernetes-backend/config.d.ts#L76-L87

 Otherwise people might start using the new parameter in their app-config and find that `yarn backstage-cli config:check --strict` starts failing.

**Comment by armandocomellas1** [comment]
> Changeset looks better and unit tests look pretty good; can you also update the config schema here to mention this new parameter?
> 
> https://github.com/backstage/backstage/blob/2d59a19fd3a566dbb8f1639aab8c79f84db7db2a/plugins/kubernetes-backend/config.d.ts#L76-L87
> 
> Otherwise people might start using the new parameter in their app-config and find that `yarn backstage-cli config:check --strict` starts failing.

Done!, already uploaded!

**Review by jamieklassen — APPROVED** [review]

**Comment by armandocomellas1** [comment]
Hi guys,

I just wondering if you have reviwed the last changes on this PR @jamieklassen @mclarke47 @Rugvip 

Thanks!

**Comment by jamieklassen** [comment]
> Hi guys,
> 
> I just wondering if you have reviwed the last changes on this PR @jamieklassen @mclarke47 @Rugvip
> 
> Thanks!

Yep, you can see I approved yesterday. If you rebase on master to include #22297, the CI checks should be fixed.

**Comment by armandocomellas1** [comment]
> > Hi guys,
> > I just wondering if you have reviwed the last changes on this PR @jamieklassen @mclarke47 @Rugvip
> > Thanks!
> 
> Yep, you can see I approved yesterday. If you rebase on master to include #22297, the CI checks should be fixed.

HI @jamieklassen,

Yes, l did already that, and the checks now are in green.

My question is, this is not going to be merged to the backstage:master branch?

Thank you!

**Comment by jamieklassen** [comment]
> I wonder if this is not going to be merged to the backstage:master branch guys?
> 
> Please let me know if you need anything else.
> 
> Thank you!

Because the PR includes docs changes (and the docs are not owned by the Kubernetes maintainer group), I can't merge this with my approval alone. This will need an approval from the core maintainers before it can be merged, and they have a lot of other PRs to consider, but I'm sure they'll take a look when they get a chance!

**Review by freben — APPROVED** [review]
Approving to unblock!

**Pull request merged**
